### PR TITLE
Fix warnings (GCC and Clang)

### LINKED ISSUE #937: "Clang" compiler produces many useful warnings
Trying clang compiler instead of gcc could help us to find problems in our code.

With the CMake build system, it's easy to use clang:

```
cd build ..
cmake -DCMAKE_C_COMPILER=clang ..
make
```

It will be beneficial to analyze the output and fix some of the warnings.

### PR DESCRIPTION
**Pull request by evgenyz**
This will also fix #937 (as a long-term solution we should have a CI with GCC + Clang builds guarded by -Werror).

By the way: @jan-cerny, @mildas #1355. Guys, really?

The only warning left is filed under #1558 (fix would break API).

### CODE REVIEW

File: src/OVAL/public/oval_types.h [hunk 1]
```diff
@@ -120,57 +120,43 @@ typedef enum {
...
-} oval_subtype_t;
+	OVAL_SUBTYPE_SYSINFO = 2,
 
 /// AIX subtypes
-typedef enum {
-	OVAL_AIX_FILESET = 1001,
+	OVAL_AIX_FILESET = OVAL_FAMILY_AIX + 1,
 	OVAL_AIX_FIX = OVAL_FAMILY_AIX + 2,
-	OVAL_AIX_OSLEVEL = OVAL_FAMILY_AIX + 3
-} oval_aix_subtype_t;
```
[the remark this answers is not in the document]
Comment (evgenyz) [reply src/OVAL/public/oval_types.h:131]: These types are not used even in our code. I have no idea at all why they were introduced.

[the remark this answers is not in the document]
Comment (evgenyz) [reply src/OVAL/public/oval_types.h:131]: To answer your question: these types are not present in any function declaration.

**Comment by evgenyz** [comment]
Rebased to the latest maint-1.3 for working CI.

File: src/OVAL/public/oval_types.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (evgenyz) [reply src/OVAL/public/oval_types.h:131]: So, the `abi-compliance-checker` doesn't mind having this change, and I can't see any sane way for these types to be used by the client code. I think that it is safe to treat it as a non-breaking change.

[the remark this answers is not in the document]
Comment (jan-cerny) [reply src/OVAL/public/oval_types.h:131]: OK, @evgenyz  thanks for executing ABI compliance checker. If it doesn't report anything, then the change should be OK. I assume that rpmdiff or other distribution checks might report something, but we can try to waive it with your explanation.

**Pull request merged**